crates/noirc_evaluator/src/ssa/acir_gen/internal_var_cache.rs

```diff
@@ -31,8 +41,26 @@ impl InternalVarCache {
 
         let mut var = match ctx.try_get_node(id)? {
             NodeObject::Const(c) => {
+                // use the InternalVar from constants if exists
                 let field_value = FieldElement::from_be_bytes_reduce(&c.value.to_bytes_be());
-                InternalVar::from_constant(field_value)
+                let mut result = InternalVar::from_constant(field_value);
+                if let Some(c_var) = self.constants.get(&field_value) {
+                    result = c_var.clone();
+                }
+
+                //use witness from other nodes if exists
+                let new_vec = Vec::new();
+                let constant_ids = ctx.constants.get(&field_value).unwrap_or(&new_vec);
```

Comment:
```suggestion
                // use witness from other nodes if exists
                let constant_ids = ctx.constants.get(&field_value).unwrap_or_default();
```

We don't need `new_vec` here.

Reply:
That what I tried first but then it complains that it cannot have a default for &Vec